projects/core/src/cms/store/selectors/component.selectors.ts

```diff
@@ -59,3 +88,84 @@ export const componentSelectorFactory = (
     }
   );
 };
+
+export const getComponentsState: MemoizedSelector<
+  StateWithCms,
+  ComponentsState
+> = createSelector(
+  getCmsState,
+  state => state.components
+);
+
+export const componentsContextSelectorFactory = (
+  uid: string
+): MemoizedSelector<StateWithCms, ComponentsContext> => {
+  return createSelector(
+    getComponentsState,
+    componentsState => {
+      // the whole component componentsState are empty
+      if (Object.keys(componentsState.entities).length === 0) {
+        return undefined;
+      } else {
+        return StateEntitySelectors.entitySelector(componentsState, uid);
+      }
+    }
+  );
+};
+
+export const componentsLoaderStateSelectorFactory = (
+  uid: string,
+  context: string
+): MemoizedSelector<StateWithCms, LoaderState<boolean>> => {
+  return createSelector(
+    componentsContextSelectorFactory(uid),
+    componentsContext => {
+      if (!componentsContext) {
+        return initialLoaderState;
+      }
+
+      const loaderState = componentsContext.pageContext[context];
+      return loaderState ? loaderState : initialLoaderState;
+    }
+  );
+};
+
+export const componentsContextExistsSelectorFactory = (
+  uid: string,
+  context: string
+): MemoizedSelector<StateWithCms, boolean> => {
+  return createSelector(
+    componentsContextSelectorFactory(uid),
+    state => {
+      if (!state) {
+        return false;
+      }
+
+      const loaderState = state.pageContext[context];
+      if (!loaderState) {
+        return false;
+      }
+
+      const exists = StateLoaderSelectors.loaderValueSelector(loaderState);
+      // 'exists' variable can be undefined, in which case we want to return false
+      return exists ? exists : false;
+    }
+  );
+};
+
+export const componentsSelectorFactory = (
+  uid: string,
+  context: string
+): MemoizedSelector<StateWithCms, CmsComponent> => {
+  return createSelector(
+    componentsContextSelectorFactory(uid),
```

Comment:
I would consider creating one more intermediate selector getting just component data for `uid`, it would possibly make it little bit optimal (as `state` changes with each page context / loading stuff, much more often than `state.component`)

Reply:
Done